src/main/groovy/com/pstehlik/groovy/gelf4j/net/GelfTransport.groovy

```diff
@@ -41,76 +26,29 @@ class GelfTransport {
    * @param gelfMessage The message to send
    */
   public void sendGelfMessageToGraylog(Gelf4JAppender appender, String gelfMessage) {
-    byte[] gzipMessage = gzipMessage(gelfMessage)
-
-    //set up chunked transfer if larger than maxChunkSize
-    if (appender.maxChunkSize < gzipMessage.size()) {
 
-      Integer chunkCount = (((gzipMessage.size() / appender.maxChunkSize) + 0.5) as Double).round().toInteger()
-      String messageId = generateMessageId()
-      chunkCount.times {
-        byte[] messageChunkPrefix = createChunkedMessagePart(messageId, it + 1, chunkCount)
-        Integer endOfChunk
-        Integer messagePartSize
-        if (gzipMessage.size() < ((it + 1) * appender.maxChunkSize)) {
-          endOfChunk = gzipMessage.size()
-          messagePartSize = gzipMessage.size() - (appender.maxChunkSize * it)
-        } else {
-          endOfChunk = (appender.maxChunkSize * (it + 1))
-          messagePartSize = appender.maxChunkSize
-        }
+    byte[] gzipMessage = gzipMessage(gelfMessage)
 
-        byte[] chunkedMessagPart = new byte[messageChunkPrefix.size() + messagePartSize]
-        System.arraycopy(
-          messageChunkPrefix,
-          0,
-          chunkedMessagPart,
-          0,
-          messageChunkPrefix.size()
-        )
-        System.arraycopy(
-          gzipMessage,
-          endOfChunk - messagePartSize,
-          chunkedMessagPart,
-          messageChunkPrefix.size(),
-          messagePartSize
-        )
-        Graylog2UdpSender.sendPacket(
-          chunkedMessagPart,
-          appender.graylogServerHost,
-          appender.graylogServerPort
-        )
-      }
-    } else {
-      Graylog2UdpSender.sendPacket(gzipMessage, appender.graylogServerHost, appender.graylogServerPort)
+    if (gzipMessage.size() <= appender.maxChunkSize) {
+      sendUdpPacket(gzipMessage, appender)
+      return
     }
-  }
 
-  /**
-   * Creates the bytes that identify a chunked message.
-   *
-   * @param messageId The unique identifier for this message
-   * @param sequenceNo The number of the message in the chunk sequence
-   * @param sequenceCount The total number of chunks
-   * @return The prepared byte array
-   */
-  private byte[] createChunkedMessagePart(String messageId, Integer sequenceNo, Integer sequenceCount) {
-    sequenceNo = sequenceNo -1
-    Collection ret = []
-    ret << 30.byteValue()
-    ret << 15.byteValue()
-    messageId.getBytes('ISO-8859-1').each {
-      ret << it
```

Comment:
All of the code above was cleaned up and moved into `GelfChunkBuilder` to restructure the package generation